Docs update minor fixes

stephenkostas opened the pull request:
Fixes a couple minor issues in the latest update that was just merged.
https://github.com/apostrophecms/apostrophe-documentation/issues/212
https://github.com/apostrophecms/apostrophe-documentation/issues/213

abea commented on other/field-properties.md:
```diff
@@ -17,6 +17,8 @@ Each [Schema Field Type](/tutorials/advanced-development/schema-guide.md#guide-t
+|[options](properties/options.md) | `object` | | Passed on to apos.area or apos.singleton as appropriate, specifying options for the allowable widgets and so on | [area](field-types/area.md), [singleton](field-types/singleton.md) | `widgets` |
 |[sortify](properties/sortify.md) | `boolean` | false | If true, make sort() operations on the field case-insensitive and more intuitive  | [string](field-types/string.md), [email](field-types/email.md) | |
 |[choices](properties/choices.md) | `array` | | An array of values that the user can select from with each being an object with value and label properties | [boolean](field-types/boolean.md) [select](field-types/select.md), [checkboxes](field-types/checkboxes.md) | `label`, `value`, `showFields` |
+|contextual | `boolean` | false | If true, will prevent the field from appearing in the dialog box for a widget | [select](field-types/select.md), [checkboxes](field-types/checkboxes.md) | |
```
This can be used by any field type.

stephenkostas replied:
Updated the field to universal and moved it up in the list.

boutell replied:
Not *every* field type, but any field type whose value is stored as a string. It wouldn't make sense for boolean, checkboxes (because they are stored as arrays), array, object, singleton, area, color, etc. It would be superfluous for `slug`. That leaves `string` and I guess `select`, anything else come to mind Bea?

abea replied:
Good point. Reading the code it looked like it was _possible_ on any field type. Advisable is another question, and we should make sure it's locked down to the ones that make sense. I recently ran into a case where another dev had used it to hide the `title` field (since you can't _remove_ it), but we wouldn't encourage its use for simply hiding fields.

I thought that areas/singletons could be marked as contextual since those would be editable in-context. Really they may be the only that would be possible. In reviewing all the projects on my machine, the property is only used when a) the field is an area/singleton or b) on `title` and `slug` to simply hide them when those properties are instead set in `beforeSave`. So the original documentation of it on `select` and `checkbox` was misguided, and we should really only direct people to it for areas and singletons.

The next question is whether to rename or update it in 3.x to take the field type into account.

stephenkostas replied:
Okay, so to make sure I don't mess this up, we want contextual to only be listed for areas and singletons? And we should remove it from anywhere else that it's listed as an available property (even though it may be *technically* available, we don't want people to use it)?

abea commented on other/field-types/select.md:
```diff
@@ -14,5 +14,5 @@ A single-select input field. The contents of the menu are set using the [`choice
-|widgetContols | `boolean` | false | If true, `checkbox` fields can be edited in line on the page if the field is in a widget |
-|contextual | `boolean` | false | If true, will prevent the `checkbox` field from appearing in the dialog box for a widget |
+|widgetContols | `boolean` | false | If true, `select` fields can be edited in line on the page if the field is in a widget |
+|contextual | `boolean` | false | If true, will prevent the `select` field from appearing in the dialog box for a widget |
```
We don't need this here anymore, right?

abea replied:
That should be it.

boutell commented:
it's not wrong to use `contextual` on a non-area-oriented field type. It's
just less likely to be literally true, in that we don't provide an out of
the box way to edit those other field types "in context" on a page. But,
the developer might do that themselves. It has happened.

On Tue, Jan 21, 2020 at 10:04 AM Alex Bea <notifications@github.com> wrote:

> *@abea* commented on this pull request.
> ------------------------------
>
> In other/field-properties.md
> <https://github.com/apostrophecms/apostrophe-documentation/pull/214#discussion_r369053936>
> :
>
> >  |[sortify](properties/sortify.md) | `boolean` | false | If true, make sort() operations on the field case-insensitive and more intuitive  | [string](field-types/string.md), [email](field-types/email.md) | |
>  |[choices](properties/choices.md) | `array` | | An array of values that the user can select from with each being an object with value and label properties | [boolean](field-types/boolean.md) [select](field-types/select.md), [checkboxes](field-types/checkboxes.md) | `label`, `value`, `showFields` |
> +|contextual | `boolean` | false | If true, will prevent the field from appearing in the dialog box for a widget | [select](field-types/select.md), [checkboxes](field-types/checkboxes.md) | |
>
> Good point. Reading the code it looked like it was *possible* on any
> field type. Advisable is another question, and we should make sure it's
> locked down to the ones that make sense. I recently ran into a case where
> another dev had used it to hide the title field (since you can't *remove*
> it), but we wouldn't encourage its use for simply hiding fields.
>
> I thought that areas/singletons could be marked as contextual since those
> would be editable in-context. Really they may be the only that would be
> possible. In reviewing all the projects on my machine, the property is only
> used when a) the field is an area/singleton or b) on title and slug to
> simply hide them when those properties are instead set in beforeSave. So
> the original documentation of it on select and checkbox was misguided,
> and we should really only direct people to it for areas and singletons.
>
> The next question is whether to rename or update it in 3.x to take the
> field type into account.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe-documentation/pull/214?email_source=notifications&email_token=AAAH27OAI4CQ6JU26HJXNFLQ64FHLA5CNFSM4KJIBTJKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCSPLRKI#discussion_r369053936>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27OOKLEAXSUUBMA3OVLQ64FHLANCNFSM4KJIBTJA>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

boutell commented:
Document it. We should not discourage people from using it to hide fields
from the modal. We might note that except for areas and singletons, there
is no built in alternative way to edit those fields, but they may have
their own plan (such as setting the field programmatically from other
fields, but still being interested in Apostrophe knowing about it for
search indexing purposes).

On Tue, Jan 21, 2020 at 10:20 AM Stephen Kostas <notifications@github.com>
wrote:

> *@stephenkostas* commented on this pull request.
> ------------------------------
>
> In other/field-properties.md
> <https://github.com/apostrophecms/apostrophe-documentation/pull/214#discussion_r369063969>
> :
>
> >  |[sortify](properties/sortify.md) | `boolean` | false | If true, make sort() operations on the field case-insensitive and more intuitive  | [string](field-types/string.md), [email](field-types/email.md) | |
>  |[choices](properties/choices.md) | `array` | | An array of values that the user can select from with each being an object with value and label properties | [boolean](field-types/boolean.md) [select](field-types/select.md), [checkboxes](field-types/checkboxes.md) | `label`, `value`, `showFields` |
> +|contextual | `boolean` | false | If true, will prevent the field from appearing in the dialog box for a widget | [select](field-types/select.md), [checkboxes](field-types/checkboxes.md) | |
>
> Okay, so to make sure I don't mess this up, we want contextual to only be
> listed for areas and singletons? And we should remove it from anywhere else
> that it's listed as an available property (even though it may be
> *technically* available, we don't want people to use it)?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe-documentation/pull/214?email_source=notifications&email_token=AAAH27P75I4VM6AAHLE2AILQ64HFPA5CNFSM4KJIBTJKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCSPOY4A#discussion_r369063969>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27JLJTCSHPFNMJV3UJTQ64HFPANCNFSM4KJIBTJA>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

stephenkostas commented:
contextual is now listed in the main properties table as a universal property, and listed alongside the other universal fields in the table for each field type.

stephenkostas commented:
Okay:
updated the text to remove the "for a widget" in the description for contextual
updated the sortify article to describe how it works for any field which has its value stored as a string
updated the listing to complete the list of sortify-able fields
